pkg/logging/log.go

```diff
@@ -0,0 +1,74 @@
+package logging
+
+import (
+	"errors"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zapr"
+	"go.uber.org/zap"
+	"k8s.io/klog/v2"
+	"k8s.io/klog/v2/klogr"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	// JSONFormat represents JSON logging mode.
+	JSONFormat = "json"
+	// TextFormat represents text logging mode.
+	// Default logging mode is TextFormat.
+	TextFormat = "text"
+)
+
+// Setup configures the logger with the supplied log format.
+// It returns an error if the JSON logger could not be initialized or passed logFormat is not recognized.
+func Setup(logFormat string) error {
+	klog.InitFlags(nil)
```

Comment:
cannot be there, has to be called before flags are parsed.

Reply:
```golang
func inti() {
	klog.InitFlags(nil)
}
```

May work ?

Reply:
@yimikao ^^

Reply:
I added this to logging package should I replace it with init()
```
func InitKlogFlags(flags *flag.FlagSet) {
	klog.InitFlags(flags)
}
```

Reply:
Yes, this way it does not need to be explicitly called.

Reply:
Not sure but I feel like you need to add this code too:
```golang
	// clear flags initialized in static dependencies
	if flag.CommandLine.Lookup("log_dir") != nil {
		flag.CommandLine = flag.NewFlagSet(os.Args[0], flag.ExitOnError)
	}
```

Reply:
Done ✅

Reply:
> @yimikao linter errors seem to be fixed now, could have been caused by non standard comment format or something like this.
> 
> Almost good but `klog.Initflags` cannot be set in `SetupLogger`.
> 
> The startup sequence has to be:
> 
>     1. init flags
> 
>     2. parse flags
> 
>     3. setup logger

Yeah, learning new things about the codebase :)

Reply:
i diidnt know init() functions are still used, thought it has been deprecated or something

Reply:
Not aware about such deprecation.

Reply:
Makes the linter unhappy :(
Go for a standard func then... we can always change later.

BTW, linter errors are back.

Reply:
Ah, seen, i'd fix